clients/src/main/java/org/apache/kafka/clients/admin/ListShareGroupOffsetsResult.java

```diff
@@ -42,6 +42,15 @@ public class ListShareGroupOffsetsResult {
             .collect(Collectors.toMap(e -> e.getKey().idValue, Map.Entry::getValue));
     }
 
+    public static ListShareGroupOffsetsResult createListShareGroupOffsetsResult(Map<String, KafkaFuture<Map<TopicPartition, Long>>> groupOffsets) {
```

Comment:
I would put this in `AdminClientTestUtils` just like the static `listConsumerGroupOffsetsResult` method which already exists. That seems to be the neatest solution to me.